ext/date/php_date.c

```diff
@@ -2525,16 +2525,13 @@ PHPAPI bool php_date_initialize_from_ts_double(php_date_obj *dateobj, double ts)
 	zend_long sec;
 	int usec;
 
-	if (UNEXPECTED(isnan(sec_dval)
-		|| sec_dval >= (double)TIMELIB_LONG_MAX
-		|| sec_dval < (double)TIMELIB_LONG_MIN
-	)) {
+	if (UNEXPECTED(isnan(sec_dval) || !PHP_DATE_DOUBLE_FITS_LONG(sec_dval))) {
 		zend_throw_error(
 			date_ce_date_range_error,
-			"Seconds must be a finite number between " TIMELIB_LONG_FMT " and " TIMELIB_LONG_FMT ", %g given",
+			"Timestamp must be a finite number between " TIMELIB_LONG_FMT " and " TIMELIB_LONG_FMT ".999999, %g given",
```

Comment:
This should probably use our default `zend_argument_error` format:

> Argument #1 ($timestamp) must be a finite number between ...

Reply:
I thought it's better matching the DateRaneError  event though the documentation needs to be updated. https://www.php.net/manual/en/class.daterangeerror.php

What do you think?

Reply:
Now I got it. This makes sense :+1: 
-> Updated the PR accordingly